694.py

```diff
@@ -0,0 +1,45 @@
+class Solution:
+    DIRS = {
+        (1, 0) : 'D',
+        (-1, 0): 'U',
```

Comment:
Do we really a dictionary here? A tuple of three elements `(dr, dc, dv)` would work well.